src/esm_parser/parse_it.py

```diff
@@ -0,0 +1,254 @@
+            node.start_mark.line,
+            node.start_mark.column,
+            yaml_file_path,
+            category,
+        )
+
+        return (data, provenance)
+
+
+class EnvironmentRepresenter(ruamel.yaml.RoundTripRepresenter):
+    """When dumping, this class is used to remove the !ENV tag from a particular value."""
+
+    def represent_scalar(self, tag, value, style=None, anchor=None):
+        type_tags = {
+            "int": f"tag:{YAML_SPEC_TAG}:int",
+            "float": f"tag:{YAML_SPEC_TAG}:float",
+            "bool": f"tag:{YAML_SPEC_TAG}:bool",
+            "str": f"tag:{YAML_SPEC_TAG}:str",
+        }
+        if tag == "!ENV":
+            type_tag = type(value).__name__
+            actual_type = type_tags[type_tag]
+            return super().represent_scalar(actual_type, value, style, anchor=anchor)
+        return super().represent_scalar(tag, value, style, anchor=anchor)
+
+
+class EsmToolsLoader(ruamel.yaml.YAML):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.filename = None
+        self.add_comments = True
+        self.env_variables = ['hallo']
+        self.Constructor = ProvenanceConstructor
+        self.Constructor.add_constructor(None, ProvenanceConstructor.construct_scalar)
+        self.Constructor.add_constructor(
+            "tag:yaml.org,2002:bool", ProvenanceConstructor.construct_yaml_bool
+        )
+        self.Representer = EnvironmentRepresenter
+
+    def get_filename(self):
+        return self.filename
+
+    def set_filename(self, filename):
+        self.filename = filename
+
+    def load(self, stream):
+        self.set_filename(stream.name)
+        mapping_with_tuple_prov = super().load(stream)[0]
+
+        def _extract_dict(mapping_with_prov):
+            config = { }
+            config_prov = { }
+            for (key,key_prov), (value,value_prov) in mapping_with_prov.items():
+                if isinstance(value, dict):
+                    config[key], config_prov[key] = _extract_dict(value)
+                elif isinstance(value, list):
+                    config[key] = []
+                    config_prov[key] = []
+                    for (elem,elem_prov) in value:
+                        prov = { }
+                        prov["line"], prov["col"], prov["yaml_file"], prov["category"] = elem_prov
+                        if isinstance(elem, dict):
+                            config[key].append(_extract_dict(elem)[0])
+                            config_prov[key].append(_extract_dict(elem)[1])
+                        else:
+                            config_prov[key].append(prov)
+                            config[key].append(elem)
+                else:
+                    prov = { }
+                    prov["line"], prov["col"], prov["yaml_file"], prov["category"] = value_prov
+                    config_prov[key] = prov
+                    config[key] = value
+            return (config, config_prov)
```

Comment:
I would suggest this is a method of the EsmToolsLoader, and not nested in `load`.

Reply:
Done